src/lib.rs

```diff
@@ -553,231 +561,45 @@ impl State {
         State {
             queue: ConcurrentQueue::unbounded(),
             local_queue: ThreadLocal::new(),
-            notified: AtomicBool::new(true),
-            sleepers: Mutex::new(Sleepers {
-                count: 0,
-                wakers: Vec::new(),
-                free_ids: Vec::new(),
-            }),
+            new_tasks: Event::new(),
             active: Mutex::new(Slab::new()),
         }
     }
 
     /// Notifies a sleeping ticker.
     #[inline]
     fn notify(&self) {
-        if self
-            .notified
-            .compare_exchange(false, true, Ordering::AcqRel, Ordering::Acquire)
-            .is_ok()
-        {
-            let waker = self.sleepers.lock().unwrap().notify();
-            if let Some(w) = waker {
-                w.wake();
-            }
-        }
+        self.new_tasks.notify(1);
     }
-}
-
-/// A list of sleeping tickers.
-struct Sleepers {
-    /// Number of sleeping tickers (both notified and unnotified).
-    count: usize,
 
-    /// IDs and wakers of sleeping unnotified tickers.
-    ///
-    /// A sleeping ticker is notified when its waker is missing from this list.
-    wakers: Vec<(usize, Waker)>,
+    /// Run a tick using the provided function to get the next task.
+    async fn tick_with(&self, mut local_ticker: impl FnMut(bool) -> Option<Runnable>) -> Runnable {
```

Comment:
imo it would make sense to change the signature of `local_ticker` such that it also takes a (mutable) reference to a resolved `self.local_queue` (i.e. the variable `local` below)

Reply:
Fixed